src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/GameManager.java

```diff
@@ -980,6 +1133,47 @@ public List<GameboardItem> getNextQuestionsForFilter(final GameFilter gameFilter
     return selectionOfGameboardQuestions;
   }
 
+  /**
+   * This generates a set of question in random.
+   *
+   * @param gameFilter - to enable search
+   * @param limit      - to provide a limit of questions
+   * @return a list of questions
+   * @throws ContentManagerException - if there is a problem accessing the content repository.
+   */
+  public List<IsaacQuestionPageDTO> generateRandomQuestions(final GameFilter gameFilter, final int limit)
+      throws ContentManagerException {
+    // 4 is the 'magic' number that limit is multiplied to make sure we have enough questions to return
```

Comment:
"Magic number" is just the name of a code smell. It means that it is a number that has no clear purpose, that it's hard to know why that number was chosen. Since you've set this number in a variable with a reasonable and informative name, it is no longer a magic number, so this comment should be edited.